tests/rados/test_bluestoretool_workflows.py

```diff
@@ -32,155 +37,336 @@ def run(ceph_cluster, **kw):
+
+            out = bluestore_obj.add_db_device(
+                osd_id=osd_id, new_device=db_target, db_size=db_size
+            )
+            log.info(out)
+            assert "DB device added" in out
+            osd_metadata = ceph_cluster.get_osd_metadata(
+                osd_id=int(osd_id), client=client
+            )
+            log.debug(f"OSD metadata for osd.{osd_id}: \n {osd_metadata}")
+
+            if not int(osd_metadata["bluefs_dedicated_db"]) == 1:
+                log.error("'bluefs_dedicated_db' entry in OSD metadata is not 1")
+                raise AssertionError(
+                    "'bluefs_dedicated_db' entry in OSD metadata is not 1"
+                )
+
+            if not (
+                osd_metadata["bluefs_db_dev_node"]
+                == osd_metadata["bluefs_db_partition_path"]
+            ):
+                log.error(
+                    "bluefs DB device node and bluefs DB partition path do not match"
+                )
+                raise AssertionError(
+                    "bluefs DB device node and bluefs DB partition path do not match"
+                )
+
+            if not osd_metadata["bluefs_db_devices"] in empty_devices[0]:
+                log.error(f"bluefs DB device is not pointing to {empty_devices[0]}")
+                raise AssertionError(
+                    f"bluefs DB device is not pointing to {empty_devices[0]}"
+                )
+
+            log.info(
+                f"Dedicated DB device successfully added for OSD {osd_id} on {empty_devices[0]}"
+            )
+
+            # Migrating new DB device to existing collocated OSD
+            # ceph-bluestore-tool bluefs-bdev-migrate --dev-target {new_device} --devs-source {device_source}
+            # Not part of execution due to blocker BZ-2269101
+            if not True:
+                log.info(
+                    f"\n -----------------------------------------------"
+                    f"\n Migrating a dedicated DB device for OSD.{osd_id}"
+                    f"\n -----------------------------------------------"
+                )
+                existing_db_device = f"dev/{osd_metadata['bluefs_db_devices']}"
+                empty_devices = rados_obj.get_available_devices(
+                    node_name=osd_host.hostname, device_type="hdd"
+                )
+                log.info(
+                    f"List of available devices on osd host {osd_host.hostname}: {empty_devices}"
+                )
+                log.info(f"Existing DB will be migrated to device: {empty_devices[0]}")
+                out = bluestore_obj.block_device_migrate(
+                    osd_id=osd_id,
+                    new_device=empty_devices[0],
+                    device_source=existing_db_device,
+                )
+                log.info(out)
+                log.debug(f"OSD metadata for osd.{osd_id}: \n {osd_metadata}")
+        else:
+            # Execute ceph-bluestore-tool --help
```

Comment:
Please add a comment here, even a debug statement should be fine, but mention that colocated scenarios are started here.

Reply:
Are not the scenarios present below not a use case for non-collocated OSDs?